crates/solver/src/solver/score_computation.rs

```diff
@@ -0,0 +1,169 @@
+// Suppose a solver generates a solution, and we compute the three variables:
+// surplus, fees and gas_cost. Currently, the objective is then simply equal to
+// surplus + fees - gas_cost. However, for CIP-20, we now have one additional
+// step for each solver. So for each solver we have trained a revert risk model,
+// which essentially is a function that computes the probability of the
+// settlement not reverting, and is a function of 3 parameters.
+// p (num_orders_in_sol, gas_used, gas_price).
+// So we compute this p, and then the score is equal to:
+// score = p * (surplus + fees) - gas_cost
+
+use {
+    super::SolverType,
+    crate::objective_value,
+    anyhow::{Context, Result},
+    clap::{Parser, ValueEnum},
+    ethcontract::U256,
+    num::ToPrimitive,
+    std::{
+        collections::HashMap,
+        fmt::{Display, Formatter},
+        ops::Neg,
+        str::FromStr,
+    },
+};
+
+#[derive(Debug, Default, Clone)]
+pub struct ScoreCalculator {
+    gas_amount_factor: f64,
+    gas_price_factor: f64,
+    nmb_orders_factor: f64,
+    intercept: f64,
+}
+
+impl ScoreCalculator {
+    pub fn new(
+        gas_amount_factor: f64,
```

Comment:
To me it feels better to make the fields pub and not have a constructor in this case. Up to you.

Reply:
Yes, no real need to be private here.